trailblazer/store/base.py

```diff
@@ -29,24 +30,54 @@ def get_job_query_with_name_and_count_labels(self) -> Query:
             func.count(Job.id).label("count"),
         )
 
-    def get_analyses_query_by_search_term_and_is_visible(
-        self,
-        search_term: str | None = None,
-        is_visible: bool = False,
-    ) -> Query:
-        """Return analyses by search term qnd is visible."""
-        if not search_term and not is_visible:
-            return
-        filter_map: dict[Callable, str | bool] | None = {
-            AnalysisFilter.FILTER_BY_IS_VISIBLE: is_visible,
-            AnalysisFilter.FILTER_BY_SEARCH_TERM: search_term,
-        }
-        filter_functions: list[Callable] = [
-            function for function, supplied_arg in filter_map.items() if supplied_arg
-        ]
-        analyses: Query = apply_analysis_filter(
-            filter_functions=filter_functions,
-            analyses=self.get_query(Analysis),
-            search_term=search_term,
-        )
-        return analyses.order_by(Analysis.started_at.desc())
+    def get_analyses_query_by_search(self, analyses: Query, search_term: str) -> Query:
+        """Return analyses by search term."""
+        if search_term:
+            analyses: Query = apply_analysis_filter(
+                filter_functions=[AnalysisFilter.FILTER_BY_SEARCH_TERM],
+                analyses=analyses,
+                search_term=search_term,
+            )
+        return analyses
+
+    def get_analyses_query_by_pipeline(self, pipeline: str) -> Query:
+        """Return analyses by pipeline."""
+        analyses: Query = self.get_query(Analysis)
+        # Group existing variants of balsamic
+        if pipeline == "balsamic":
+            analyses = analyses.filter(Analysis.data_analysis.like("%balsamic%"))
+        if pipeline:
+            analyses = analyses.filter(Analysis.data_analysis == pipeline)
+        return analyses.filter(Analysis.is_visible.is_(True))
+
+    def filter(self, analyses: Query, query: AnalysisRequest) -> Query:
+        if query.status:
+            analyses = analyses.filter(Analysis.status.in_(query.status))
+        if query.priority:
+            analyses = analyses.filter(Analysis.priority.in_(query.priority))
+        if query.type:
+            analyses = analyses.filter(Analysis.type.in_(query.type))
+        if query.comment:
+            analyses = analyses.filter(or_(Analysis.comment.is_(None), Analysis.comment == ""))
+        return analyses
+
+    def sort(self, analyses: Query, query: AnalysisRequest) -> Query:
+        if query.sort_field:
+            column = getattr(Analysis, query.sort_field)
+            order_function = asc if query.sort_order == "asc" else desc
```

Comment:
A type hint here would really help with the intention!

Reply:
In this instance it would not help that much since it is a function, but I refactored it